src/js/tabs/settingsgateway.js

```diff
@@ -124,6 +126,40 @@ SettingsGatewayTab.prototype.angular = function(module)
             });
           }
           break;
+        case 'requireAuthFlag':
+          // Need to set flag on account_root only when chosen option is different from current setting
+          if ($scope.currentRequireAuthFlagSetting !== $scope.isRequireAuthFlagEnabled) {
+            $scope.edit.requireAuthFlagSaving = true;
+            var tx = network.remote.transaction();
+            !$scope.isRequireAuthFlagEnabled ? tx.accountSet(id.account, undefined, Transaction.set_clear_flags.AccountSet.asfRequireAuth) : tx.accountSet(id.account, Transaction.set_clear_flags.AccountSet.asfRequireAuth);
```

Comment:
Looks like this logic was originally added by darkdarkdragon, but in my mind it is confusing, unless you know the ripple-lib source code.  I think a comment would help make it clearer.